### PR TITLE
enable linkcheck in doc build

### PR DESCRIPTION
**Pull request by guotuofeng**
## Describe your changes
enable linkcheck in doc build
## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [x] Update documents if necessary.
- [x] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

**Review by jambayk — COMMENTED** [review]
How do we handle cases where some new files are being created in the same PR as the one adding new links? In this case the links to the new files won't exist at the time of the PR. 

I think it might be better to create a new pipeline that is triggered after a PR merges and only runs `make linkcheck`. This way we get a status of the links but it doesn't block the PR validation pipelines or the doc publish pipeline. 

So, it will be similar to this pipeline https://github.com/microsoft/Olive/blob/main/.azure_pipelines/build-doc.yaml.

**Comment by guotuofeng** [comment]
> How do we handle cases where some new files are being created in the same PR as the one adding new links? In this case the links to the new files won't exist at the time of the PR.
> 
> I think it might be better to create a new pipeline that is triggered after a PR merges and only runs `make linkcheck`. This way we get a status of the links but it doesn't block the PR validation pipelines or the doc publish pipeline.
> 
> So, it will be similar to this pipeline https://github.com/microsoft/Olive/blob/main/.azure_pipelines/build-doc.yaml but only for linkcheck

It seems there are not ideal solution for your mentioned scenario. Another solution is to add the file firstly in a PR, then update the link in following PR, which is same with your suggestion in theory.

**Review by trajepl — APPROVED** [review]

**Pull request merged**
